pkg/generator/password_reconciler.go

```
@@ -115,37 +123,96 @@ func (r *PasswordReconciler) createSecret(ctx context.Context, password *sgv1alp
 	return reconcile.Result{}, nil
 }
 
-type passwordParams struct {
-	Length int `yaml:"length"`
+func (r *PasswordReconciler) generate(password *sgv1alpha1.Password) (string, error) {
+	passwordVal := localGeneratePassword(&password.Spec)
+	return passwordVal, nil
 }
 
-func (r *PasswordReconciler) generate(password *sgv1alpha1.Password) (string, error) {
-	gen := cfgtypes.NewPasswordGenerator()
+func localGeneratePassword(spec *sgv1alpha1.PasswordSpec) string {
+
+	var password strings.Builder
+
+	//Set symbol character
+	for i := 0; i < spec.Symbols; i++ {
+		value, _ := randomElement(spec.SymbolCharSet)
```

seems like we we are ignoring errors when generating elements, i think they should be propagated since failure to generate char shouldnt be ignored.